### PR TITLE
feature/file-transaction-svc.proto

### PR DESCRIPTION
**Pull request by dinhl**
This is the proto or contract for the hwsc-file-transaction-svc

### CODE REVIEW

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 1]
```diff
@@ -2,23 +2,28 @@ syntax="proto3";
...
 service FileTransactionService{
     rpc GetStatus(FileTransactionRequest) returns (FileTransactionResponse) {}
-}
-
-enum State {
-    AVAILABLE = 0;
-    UNAVAILABLE = 1;
-    SUCCESS = 2;
-    FAIL = 3;
+    rpc UploadFile(stream FileTransactionRequest)returns(FileTransactionResponse){}
```
Comment (faraonc) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: we dont want to stream FileTransactionRequest. This is too heavy

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 2]
```diff
@@ -2,23 +2,28 @@ syntax="proto3";
...
+    rpc DownloadFile(FileTransactionRequest)returns(stream FileTransactionResponse){}
```
Comment (faraonc) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: we dont want to stream FileTransactionResponse. this is too heavy

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 3]
```diff
@@ -2,23 +2,28 @@ syntax="proto3";
...
 }
 
 message FileTransactionRequest{
     string message = 1;
+    string name =2;
+    bytes buffer =3;
```
Comment (faraonc) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: consider refactoring buffer into a different message such as Segment

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 4]
```diff
@@ -2,23 +2,28 @@ syntax="proto3";
...
+    string url =4;
 }
 
 message FileTransactionResponse{
-    State state = 1;
+    oneof status{uint32 code =1;}
+    string message =2;
+    bytes buffer =3;
```
Comment (faraonc) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: consider refactoring buffer into a different message such as Segment

Comment (faraonc) [reply int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: add int64 length field

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto (hunk 1 above)
Comment (faraonc) [reply int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: Segment is what we were streaming right?

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 5]
```diff
@@ -2,23 +2,31 @@ syntax="proto3";
...
+    string name =2;
+    string url =3;
 }
 
 message FileTransactionResponse{
-    State state = 1;
+    oneof status{uint32 code =1;}
+    string message =2;
+    int64 length =3;
+    string url =4;
```
Comment (faraonc) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: switch url as 3 and length as 4

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 6]
```diff
@@ -2,23 +2,31 @@ syntax="proto3";
...
 }
+message Chunk{
+    bytes buffer =1;
+}
+
+
```
Comment (faraonc) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: fix spacing and indentation

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 7]
```diff
@@ -2,23 +2,25 @@ syntax="proto3";
...
 service FileTransactionService{
     rpc GetStatus(FileTransactionRequest) returns (FileTransactionResponse) {}
-}
-
-enum State {
-    AVAILABLE = 0;
-    UNAVAILABLE = 1;
-    SUCCESS = 2;
-    FAIL = 3;
+    rpc UploadFile(stream Chunk)returns(FileTransactionResponse){}
```
Comment (faraonc) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: spacing in between returns

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 8]
```diff
@@ -2,23 +2,25 @@ syntax="proto3";
...
+    rpc DownloadFile(FileTransactionRequest)returns(stream Chunk){}
```
Comment (faraonc) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: spacing in between returns

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 9]
```diff
@@ -2,23 +2,25 @@ syntax="proto3";
...
 }
 
 message FileTransactionRequest{
```
Comment (faraonc) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: spacing in between assignment operators

File: int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto [hunk 10]
```diff
@@ -1,24 +1,26 @@
...
+    string url = 3;
 }
 
 message FileTransactionResponse{
-    State state = 1;
+    oneof status {uint32 code = 1;}
+    string message = 2;
+    string url = 3;
+    int64 length = 4;
 }
```
Comment (wai-swk) [inline int/hwsc-file-transaction-svc/proto/hwsc-file-transaction-svc.proto]: Add a new line after this line for uniformity maybe?

**Comment by faraonc** [comment]
great job

**Pull request merged**